prompting/base/validator.py

```diff
@@ -297,18 +297,17 @@ def resync_metagraph(self):
             "Metagraph updated, re-syncing hotkeys, dendrite pool and moving averages"
         )
         # Zero out all hotkeys that have been replaced.
-        for uid, hotkey in enumerate(self.hotkeys):
-            if hotkey != self.metagraph.hotkeys[uid]:
-                self.scores[uid] = 0  # hotkey has been replaced
+        min_len = min(len(self.hotkeys), len(self.metagraph.hotkeys))
+        for uid in range(min_len):
```

Comment:
Since we have no idea how a metagraph is lost from a hotkey, can we always guarantee that it is the last uid that got deleted?

Reply:
> Since we have no idea how a metagraph is lost from a hotkey, can we always guarantee that it is the last uid that got deleted?

That's a good point. Quickest workaround would be to search which uid was lost exactly, and update scores accordingly

Reply:
@bkb2135 updated this part of code, with the assumption that elements of `metagraph.hotkeys` are unique, as stated in docs [[1]](https://docs.bittensor.com/subnets/working-with-keys#hotkey). So that we can map each hotkey to an index:
```
hotkey_to_index = {hotkey: idx for idx, hotkey in enumerate(self.hotkeys)}
```

Reply:
This change is excluded from this PR